openedx/core/lib/request_utils.py

```diff
@@ -153,54 +153,61 @@ def process_request(self, request):
                 # Add or update the size for this group.
                 cookie_groups_to_size[grouping_name] = cookie_groups_to_size.get(grouping_name, 0) + cookie_size
 
-        max_cookie_name = max(cookie_names_to_size, key=lambda name: cookie_names_to_size[name])
-        max_cookie_size = cookie_names_to_size[max_cookie_name]
+        if cookie_names_to_size:
```

Comment:
**Reviewer Note:** The actual fix was this (and the following) if-condition.
When moving code around, I also decided to extract `set_custom_attributes_for_top_n`.  This was not needed for the fix, but just shrank this function a little and removed some of the redundancy.